sirepo/package_data/static/js/srw.js

```diff
@@ -807,6 +807,82 @@ SIREPO.app.controller('BeamlineController', function (activeSection, appState, b
     });
 });
 
+SIREPO.app.controller('MLController', function (appState, panelState, persistentSimulation, requestSender, srwService, $scope, $window) {
+    const self = this;
+    self.appState = appState;
+    self.srwService = srwService;
+    self.simScope = $scope;
+    self.resultsFile = null;
+    self.simComputeModel = 'machineLearningAnimation';
+    self.simState = persistentSimulation.initSimulationState(self);
+
+    function openActivait(data) {
+        requestSender.sendStatelessCompute(
+            appState,
+            d => {
+                $window.location.href = `activait#/data/${data.models.simulation.simulationId}`;
```

Comment:
This doesn't seem right at all. Why is there a hardwired URL? There are routines for formatting URLs and local redirects.

Reply:
That's where I started, but nothing gets you from one app to another app with a specific sim id - at best you can reach the sim list with `globalRedirect()`. I used the srw -> shadow conversion as a guide:
https://github.com/radiasoft/sirepo/blob/8fb85dc7befa7684ac4de4631b053c75d47a014a/sirepo/package_data/static/js/sirepo-components.js#L3036
which only provides a link of this same format for the user to click. I think an issue to cover this case in requestSender is warranted.

Reply:
If you see something that's "not right", then it's time to fix it, not copy it. Operations like this should always be server side, similar to the problem. For example #1972 is still a problem. The reason it isn't fixed is that it takes time. The GUI should not be responsible for operations like this. Please add an issue that the SRW-Shadow conversion code should be server-side, too.

> nothing gets you from one app to another app with a specific sim id

This is a bug, not a feature (or a fact). Our software is one ecosystem, there are no walls that prevent you from modifying the code to be better. If you have the time, and it seems to me you do, then please do it right. It's not complicated or complex.

Reply:
I don't always have a good feel for this. If it's in the code, and not egregiously weird, then my assumption is that it is "right". It is, at least, "how we do it". I'll use this experience to expand what I regard as "weird".

At the time, I had no time, but at this time I have the time

Reply:
That's fair. A skill to improve. Here's a simple rule: If a server response to the GUI is to turn around and do something else immediately (not after a poll, like runStatus) on the server, that's weird. Why didn't the server just do both things? This has to do with data locality. Why would the data come all the way back just to be pushed to the same server again. 

Another rule: don't give the server something it already has. A good example of this is runSimulation. The data should already be saved on the server when runSimulation is called. Otherwise, there will be a collision with the data on disk vs what simulation is being run. Therefore, the only thing that should be sent is the identity of the data (including version/serial) to runSimulation.

Reply:
[93f26d2](https://github.com/radiasoft/sirepo/pull/5216/commits/93f26d2abc840faddd2fc8d2684fad38c0908736)